src/devtools/client/debugger/src/components/TestInfo/ContextMenu.tsx

```diff
@@ -0,0 +1,119 @@
+import { useContext, useRef } from "react";
+
+import useModalDismissSignal from "bvaughn-architecture-demo/src/hooks/useModalDismissSignal";
+import { ReplayClientContext } from "shared/client/ReplayClientContext";
+import { seekToTime, startPlayback } from "ui/actions/timeline";
+import MaterialIcon from "ui/components/shared/MaterialIcon";
+import { useAppDispatch, useAppSelector } from "ui/setup/hooks";
+
+import { selectLocation } from "../../actions/sources";
+import { getContext } from "../../selectors";
+import {
+  Coordinates,
+  TestCaseType,
+  TestInfoContextMenuContext,
+  TestStepType,
+} from "./TestInfoContextMenuContext";
+import styles from "./ContextMenu.module.css";
+
+function ContextMenu({
+  hide,
+  mouseCoordinates,
+  testStep,
+  testCase,
+}: {
+  hide: () => void;
+  mouseCoordinates: Coordinates;
+  testStep: TestStepType;
+  testCase: TestCaseType;
+}) {
+  const dispatch = useAppDispatch();
+  const ref = useRef<HTMLDivElement>(null);
+  const cx = useAppSelector(getContext);
+  const client = useContext(ReplayClientContext);
+
+  useModalDismissSignal(ref, hide, true);
+
+  const onPlayFromHere = () => {
+    hide();
+    dispatch(startPlayback({ beginTime: testStep.startTime, endTime: testCase.endTime }));
+  };
+  const onJumpToBefore = () => {
+    hide();
+    dispatch(seekToTime(testStep.startTime));
+  };
+  const onJumpToAfter = () => {
+    hide();
+    dispatch(seekToTime(testStep.endTime - 1));
+  };
+  const onGoToLocation = async () => {
+    hide();
+    if (!testStep.enqueuePoint) {
+      return;
+    }
+
+    const frame = await (async point => {
```

Comment:
This doesn't need to be wrapped in another async iife because we're already in an async fn